core/src/scene/sceneLoader.cpp

```diff
@@ -916,43 +917,52 @@ void SceneLoader::loadCameras(Node _cameras, Scene& _scene) {
     // right now, we'll just apply the settings from the first active camera we
     // find.
 
-    for (const auto& cam : _cameras) {
+    auto& view = _scene.view();
 
-        const Node camera = cam.second;
+    for (const auto& entry : _cameras) {
 
-        const std::string type = camera["type"].as<std::string>();
+        const Node camera = entry.second;
+
+        if (Node active = camera["active"]) {
+            if (!active.as<bool>()) {
+                continue;
+            }
+        }
+
+        auto type = camera["type"].Scalar();
         if (type == "perspective") {
-            // The default camera
-            Node fov = camera["fov"];
-            if (fov) {
+
+            view->setCameraType(CameraType::perspective);
+
+            if (Node fov = camera["fov"]) {
                 // TODO
             }
 
-            Node focal = camera["focal_length"];
-            if (focal) {
+            if (Node focal = camera["focal_length"]) {
                 // TODO
             }
 
-            Node vanishing = camera["vanishing_point"];
-            if (vanishing) {
+            if (Node vanishing = camera["vanishing_point"]) {
                 // TODO
             }
         } else if (type == "isometric") {
-            // TODO
-            Node axis = camera["axis"];
-            if (axis) {
-                // TODO
+
+            view->setCameraType(CameraType::isometric);
+
+            if (Node axis = camera["axis"]) {
+                view->setObliqueAxis(axis[0].as<float>(), axis[1].as<float>());
             }
         } else if (type == "flat") {
-            // TODO
+
+            view->setCameraType(CameraType::flat);
+
         }
 
         double x = -74.00976419448854;
         double y = 40.70532700869127;
         float z = 16;
```

Comment:
Map starting location can be defined by the camera, we should follow up
with that if we don't support already. And yeah having a NYC baked into
core engine code is weird :) Can put a default in the demo app itself.

On Wed, Oct 21, 2015 at 10:24 AM, Varun <notifications@github.com> wrote:

> In core/src/scene/sceneLoader.cpp
> <https://github.com/tangrams/tangram-es/pull/350#discussion_r42638555>:
>
> >              }
> >          } else if (type == "flat") {
> > -            // TODO
> > +
> > +            view->setCameraType(CameraType::flat);
> > +
> >          }
> >
> >          double x = -74.00976419448854;
> >          double y = 40.70532700869127;
> >          float z = 16;
>
> Should we take these position values out from here and set these in the
> scene.yaml for tangram-es?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/350/files#r42638555>.
>